### PR TITLE
[Wallet] Dialog Component

### PR DESCRIPTION
**Pull request by aaronmgdr**
### Description

Adds a Dialog Component for the Redesign. 

There is also a blank Modal Component which is simpler, eventually this would be where we set how modals animate. 

### Other changes

Steals Card from @jeanregisser 

#### DIALOG
Screen Shot 2020-05-21 at 10 18 54 AM
Screen Shot 2020-05-21 at 10 22 03 AM
Screen Shot 2020-05-21 at 10 32 24 AM
Screen Shot 2020-05-21 at 10 32 37 AM
Screen Shot 2020-05-21 at 10 43 22 AM

#### with really long button text
Screen Shot 2020-05-21 at 11 27 03 AM

#### MODAL
Screen Shot 2020-05-21 at 10 47 43 AM

Screen Shot 2020-05-21 at 10 47 11 AM

### Related issues

- #3645

### CODE REVIEW

File: packages/mobile/src/components/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+interface Props {
+  children: React.ReactNode
+  isVisible: boolean
+  style?: StyleProp<ViewStyle>
+}
+
+export default function Modal({ children, isVisible, style }: Props) {
+  return (
+    <ReactNativeModal isVisible={isVisible} backdropOpacity={0.1}>
+      <Card style={[styles.root, style]} rounded={true}>
```
Comment (jeanregisser) [inline packages/mobile/src/components/Modal.tsx]: We need to wrap the `Card` inside `<SafeAreaView>` so it doesn't render below the status bar or under the home indicator on iOS.

**Review by jeanregisser — CHANGES_REQUESTED** [review]
Great!! 🎉

File: packages/mobile/src/components/Modal.tsx (hunk 1 above)
Comment (aaronmgdr) [reply packages/mobile/src/components/Modal.tsx]: done

**Review by jmrossy — COMMENTED** [review]
Can you confirm with @annakaz if this is redundant with anything she did for her recent verification flow changes?

**Comment by aaronmgdr** [comment]
> Can you confirm with @annakaz if this is redundant with anything she did for her recent verification flow changes?

I asked her and she said the work she did was just deduplicating some in the old design

**Review by jeanregisser — APPROVED** [review]

**Pull request merged**